### PR TITLE
Lua filter: support encoding binary data via base64Escape

### PR DESCRIPTION
**Pull request by lhluo**
Commit Message: Support binary data in Lua Base64 encoding 
Additional Description: `base64Escape` was introduced in #12552 to allow Lua filters to Base64 encode strings. The input is stored in `string_view` which assumes a null-terminated string by default. However, raw binary data is not supported because it can have null bytes anywhere. This PR fixes the issue by setting the input_size explicitly.

Risk Level: Low
Testing: Added a test case in unit tests
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: Lihao Luo <lihao@squareup.com>

### CODE REVIEW

File: test/extensions/filters/http/lua/lua_filter_test.cc [hunk 1]
```diff
@@ -2216,9 +2218,16 @@ TEST_F(LuaHttpFilterTest, LuaFilterBase64Escape) {
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
 
   Http::TestResponseHeaderMapImpl response_headers{{":status", "200"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::StopIteration,
+            filter_->encodeHeaders(response_headers, false));
 
-  EXPECT_CALL(*filter_, scriptLog(spdlog::level::trace, StrEq("YmFyZm9v")));
-  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->encodeHeaders(response_headers, true));
```
Comment (dio) [inline test/extensions/filters/http/lua/lua_filter_test.cc:2221]: Should we keep a test with a simple string as input as well?

File: test/extensions/filters/http/lua/lua_filter_test.cc [hunk 2]
```diff
@@ -2216,9 +2218,16 @@ TEST_F(LuaHttpFilterTest, LuaFilterBase64Escape) {
...
+  // base64 encoding should also work for binary data
```
Comment (dio) [inline test/extensions/filters/http/lua/lua_filter_test.cc:2224]: Super nit. 

```
// Base64 encoding should also work for binary data.
```

**Review by dio — COMMENTED** [review]
Thanks for catching this. Few comments.

**Comment by lhluo** [comment]
@dio thanks for reviewing! Updated PR to address your comments.

**Comment by lhluo** [comment]
Force push to fix DCO

**Comment by dio** [comment]
@lhluo could you merge master? Thanks! Seems like there is a fix for that coverage failure.

**Comment by lhluo** [comment]
@dio done!

**Review by dio — APPROVED** [review]
Thanks.

**Comment by lhluo** [comment]
hi @dio could you help merge these changes today?

**Review by snowp — APPROVED** [review]
LGTM, thanks for fixing!

**Pull request merged**